types/next-auth/client.d.ts

```diff
@@ -50,21 +50,29 @@ declare function signin(
     provider: 'credentials' | 'email',
     data: Record<string, unknown> & {
         callbackUrl?: string;
-        redirect: false;
+        redirect?: true;
     },
     authorizationParams?: string | string[][] | Record<string, unknown> | URLSearchParams
-  ): Promise<SignInResponse>;
+  ): Promise<void>;
 declare function signin(
-    provider?: string,
+    provider: 'credentials' | 'email',
+    data: Record<string, unknown> & {
+        callbackUrl?: string;
+        redirect: boolean;
+    },
+    authorizationParams?: string | string[][] | Record<string, unknown> | URLSearchParams
+): Promise<SignInResponse>;
+declare function signin(
+    provider?: Omit<string, 'credentials' | 'email'>,
     data?: Record<string, unknown> & {
         callbackUrl?: string;
-        redirect?: boolean;
+        redirect?: true;
     },
     authorizationParams?: string | string[][] | Record<string, unknown> | URLSearchParams
 ): Promise<void>;
 declare const signIn: typeof signin;
-declare function signout(data: { callbackUrl?: string, redirect: false }): Promise<SignOutResponse>;
-declare function signout(data?: { callbackUrl?: string, redirect?: boolean }): Promise<void>;
+declare function signout(data: { callbackUrl?: string, redirect: boolean }): Promise<SignOutResponse>;
+declare function signout(data?: { callbackUrl?: string, redirect?: true }): Promise<void>;
 declare const signOut: typeof signout;
```

Comment:
@lluia I commited the changes, but i had to make an adjusment to the type check work:
- Added another signin for the case when you have `email` or `credentials` but with the `redirect?: true`

Reply:
@lluia llet me know when you come back
We can already merge it, but it needs to be done by you